### ISSUE TITLE
Encounter creation failed for woman and house hold head in openMRS due to modified pattern of new house hold registration and census form

### ISSUE THREAD

**Issue by sohelsarder** [OpenSRP/opensrp-server#94]
@ndisha , @maimoonak , @julkarnain, @numeranazneen 
model.xml (current):
https://github.com/OpenSRP/opensrp-server/blob/external_system_integration/assets/form/new_household_registration/model.xml

model.xml (old):
https://github.com/OpenSRP/opensrp-server/blob/master/assets/form/new_household_registration/model.xml

it needs to introduce  one level depth iteration logic in openmrs connector to get eligibles fields inside woman tag.

**Comment by maimoonak** [comment]
@sohelsarder Could you please share form submission i.e. in submitted data`s couchdb document

**Comment by sohelsarder** [comment]
@maimoonak here it is please let me know if anything else require?

{
   "_id": "5ec11f03071ef0929839079530028992",
   "_rev": "2-ddf675dd499e017538671b94843aeefc",
   "type": "FormSubmission",
   "anmId": "opensrp",
   "instanceId": "d304dbr7-3998-434f-8c5b-55d6f5fa4252",
   "formName": "new_household_registration",
   "entityId": "4237d267-d438-49f2-7822-8968t555447c",
   "clientVersion": 1444562091545,
   "formDataDefinitionVersion": "12",
   "formInstance": {
       "form_data_definition_version": "12",
       "form": {
           "bind_type": "household",
           "default_bind_path": "/model/instance/FWNewHH",
           "fields": [
               {
                   "name": "id",
                   "value": "4237d267-d438-49f2-7822-8968t555447c",
                   "source": "household.id"
               },
               {
                   "name": "location_name",
                   "value": "DASLIA_-_ALL_PARAS",
                   "source": "household.location_name"
               },
               {
                   "name": "existing_location",
                   "value": "2fc43738-ace5-g961-8e8f-ab7dg0e5bc63",
                   "source": "household.existing_location"
               },
               {
                   "name": "existing_Country",
                   "value": "Bangladesh",
                   "source": "household.existing_Country"
               },
               {
                   "name": "existing_Division",
                   "value": "RANGPUR",
                   "source": "household.existing_Division"
               },
               {
                   "name": "existing_District",
                   "value": "GAIBANDHA",
                   "source": "household.existing_District"
               },
               {
                   "name": "existing_Upazilla",
                   "value": "SADULLAPUR",
                   "source": "household.existing_Upazilla"
               },
               {
                   "name": "existing_Union",
                   "value": "NALDANGA",
                   "source": "household.existing_Union"
               },
               {
                   "name": "existing_Ward",
                   "value": "Ward-3",
                   "source": "household.existing_Ward"
               },
               {
                   "name": "existing_Subunit",
                   "value": "3-KHA",
                   "source": "household.existing_Subunit"
               },
               {
                   "name": "existing_Mauzapara",
                   "value": "DASLIA+-+ALL+PARAS",
                   "source": "household.existing_Mauzapara"
               },
               {
                   "name": "today",
                   "value": "2015-10-11",
                   "source": "household.today"
               },
               {
                   "name": "start",
                   "value": "2015-10-11T17:12:34.000+06:00",
                   "source": "household.start"
               },
               {
                   "name": "end",
                   "value": "2015-10-11T17:12:34.000+06:00",
                   "source": "household.end"
               },
               {
                   "name": "FWNHREGDATE",
                   "value": "2015-10-11",
                   "source": "household.FWNHREGDATE"
               },
               {
                   "name": "FWGOBHHID",
                   "value": "2322",
                   "source": "household.FWGOBHHID"
               },
               {
                   "name": "FWJIVHHID",
                   "value": "9889",
                   "source": "household.FWJIVHHID"
               },
               {
                   "name": "FWCOUNTRY",
                   "value": "Bangladesh",
                   "source": "household.FWCOUNTRY"
               },
               {
                   "name": "FWDIVISION",
                   "value": "RANGPUR",
                   "source": "household.FWDIVISION"
               },
               {
                   "name": "FWDISTRICT",
                   "value": "GAIBANDHA",
                   "source": "household.FWDISTRICT"
               },
               {
                   "name": "FWUPAZILLA",
                   "value": "SADULLAPUR",
                   "source": "household.FWUPAZILLA"
               },
               {
                   "name": "FWUNION",
                   "value": "NALDANGA",
                   "source": "household.FWUNION"
               },
               {
                   "name": "FWWARD",
                   "value": "Ward-3",
                   "source": "household.FWWARD"
               },
               {
                   "name": "FWSUBUNIT",
                   "value": "3-KHA",
                   "source": "household.FWSUBUNIT"
               },
               {
                   "name": "FWMAUZA_PARA",
                   "value": "DASLIA+-+ALL+PARAS",
                   "source": "household.FWMAUZA_PARA"
               },
               {
                   "name": "FWNHHHGPS",
                   "value": "23.8002 90.4068 0 10",
                   "source": "household.FWNHHHGPS"
               },
               {
                   "name": "FWHOHFNAME",
                   "value": "mango",
                   "source": "household.FWHOHFNAME"
               },
               {
                   "name": "FWHOHLNAME",
                   "value": ".",
                   "source": "household.FWHOHLNAME"
               },
               {
                   "name": "FWHOHBIRTHDATE",
                   "value": "1900-01-01",
                   "source": "household.FWHOHBIRTHDATE"
               },
               {
                   "name": "FWHOHGENDER",
                   "value": "2",
                   "source": "household.FWHOHGENDER"
               },
               {
                   "name": "FWNHHMBRNUM",
                   "value": "23",
                   "source": "household.FWNHHMBRNUM"
               },
               {
                   "name": "FWNHHMWRA",
                   "value": "1",
                   "source": "household.FWNHHMWRA"
               },
               {
                   "name": "join_names",
                   "value": "hoby",
                   "source": "household.join_names"
               },
               {
                   "name": "MWRA",
                   "value": "1",
                   "source": "household.MWRA"
               }
           ],
           "sub_forms": [
               {
                   "name": "elco_registration",
                   "bind_type": "elco",
                   "default_bind_path": "/model/instance/FWNewHH/woman",
                   "fields": [
                       {
                           "name": "id",
                           "source": "elco.id"
                       },
                       {
                           "name": "relationalid",
                           "source": "elco.relationalid"
                       },
                       {
                           "name": "WomanREGDATE",
                           "source": "elco.WomanREGDATE"
                       },
                       {
                           "name": "GOBHHID",
                           "source": "elco.GOBHHID"
                       },
                       {
                           "name": "JiVitAHHID",
                           "source": "elco.JiVitAHHID"
                       },
                       {
                           "name": "FWWOMFNAME",
                           "source": "elco.FWWOMFNAME"
                       },
                       {
                           "name": "FWWOMLNAME",
                           "source": "elco.FWWOMLNAME"
                       },
                       {
                           "name": "FWBIRTHDATE",
                           "source": "elco.FWBIRTHDATE"
                       },
                       {
                           "name": "FWGENDER",
                           "source": "elco.FWGENDER"
                       },
                       {
                           "name": "FWWOMAGE",
                           "source": "elco.FWWOMAGE"
                       },
                       {
                           "name": "display_age",
                           "source": "elco.display_age"
                       },
                       {
                           "name": "FWNHWOMSTRMEN",
                           "source": "elco.FWNHWOMSTRMEN"
                       },
                       {
                           "name": "FWNHWOMHUSALV",
                           "source": "elco.FWNHWOMHUSALV"
                       },
                       {
                           "name": "FWNHWOMHUSLIV",
                           "source": "elco.FWNHWOMHUSLIV"
                       },
                       {
                           "name": "FWNHWOMHUSSTR",
                           "source": "elco.FWNHWOMHUSSTR"
                       },
                       {
                           "name": "FWELIGIBLE",
                           "source": "elco.FWELIGIBLE"
                       },
                       {
                           "name": "FWWOMANYID",
                           "source": "elco.FWWOMANYID"
                       },
                       {
                           "name": "FWWOMNID",
                           "source": "elco.FWWOMNID"
                       },
                       {
                           "name": "FWWOMRETYPENID",
                           "source": "elco.FWWOMRETYPENID"
                       },
                       {
                           "name": "FWWOMBID",
                           "source": "elco.FWWOMBID"
                       },
                       {
                           "name": "FWWOMRETYPEBID",
                           "source": "elco.FWWOMRETYPEBID"
                       },
                       {
                           "name": "FWHUSNAME",
                           "source": "elco.FWHUSNAME"
                       },
                       {
                           "name": "FWWOMCOUNTRY",
                           "source": "elco.FWWOMCOUNTRY"
                       },
                       {
                           "name": "FWWOMDIVISION",
                           "source": "elco.FWWOMDIVISION"
                       },
                       {
                           "name": "FWWOMDISTRICT",
                           "source": "elco.FWWOMDISTRICT"
                       },
                       {
                           "name": "FWWOMUPAZILLA",
                           "source": "elco.FWWOMUPAZILLA"
                       },
                       {
                           "name": "FWWOMUNION",
                           "source": "elco.FWWOMUNION"
                       },
                       {
                           "name": "FWWOMWARD",
                           "source": "elco.FWWOMWARD"
                       },
                       {
                           "name": "FWWOMSUBUNIT",
                           "source": "elco.FWWOMSUBUNIT"
                       },
                       {
                           "name": "FWWOMMAUZA_PARA",
                           "source": "elco.FWWOMMAUZA_PARA"
                       },
                       {
                           "name": "FWWOMGOBHHID",
                           "source": "elco.FWWOMGOBHHID"
                       },
                       {
                           "name": "FWWOMGPS",
                           "source": "elco.FWWOMGPS"
                       }
                   ],
                   "instances": [
                       {
                           "GOBHHID": "2322",
                           "JiVitAHHID": "9889",
                           "WomanREGDATE": "2015-10-11",
                           "FWWOMFNAME": "jackfruit",
                           "FWWOMLNAME": ".",
                           "FWBIRTHDATE": "1988-10-08",
                           "FWGENDER": "female",
                           "FWWOMAGE": "27",
                           "display_age": "",
                           "FWNHWOMSTRMEN": "0",
                           "FWNHWOMHUSLIV": "1",
                           "FWNHWOMHUSALV": "",
                           "FWNHWOMHUSSTR": "0",
                           "FWELIGIBLE": "1",
                           "FWWOMANYID": "1 2",
                           "FWWOMNID": "7675777777775",
                           "FWWOMRETYPENID": "7675777777775",
                           "FWWOMBID": "99999998888888888",
                           "FWWOMRETYPEBID": "99999998888888888",
                           "FWHUSNAME": "zoom",
                           "FWWOMCOUNTRY": "Bangladesh",
                           "FWWOMDIVISION": "RANGPUR",
                           "FWWOMDISTRICT": "GAIBANDHA",
                           "FWWOMUPAZILLA": "SADULLAPUR",
                           "FWWOMUNION": "NALDANGA",
                           "FWWOMWARD": "Ward-3",
                           "FWWOMSUBUNIT": "3-KHA",
                           "FWWOMMAUZA_PARA": "DASLIA+-+ALL+PARAS",
                           "FWWOMGOBHHID": "2322",
                           "FWWOMGPS": "23.8002 90.4068 0 10",
                           "id": "ce71572a-8oc5-u32f-9d3b-4a6b568d5g77"
                       }
                   ]
               }
           ]
       }
   },
   "serverVersion": 1444734863350
}

**Comment by maimoonak** [comment]
@sohelsarder @cagulas  I added a test case for testing this and found that grouped data in xlsform does not impact data push into openmrs. There were few other problems found with address mappings (although those were not causing any failure) and I have fixed in the pull request referred above. 

However, below are problems I found with xlsform (resolved manually in testcase resources).

- form submission has int vale for gender i.e. (not a problem though) gender=2
- FWNHHMWRA - 1901AAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA is a numeric concept and can not have value mapped to coded data
- FWELIGIBLE - 162699AAAAAAAAAAAAAAAAAAAAAAAAAA is a coded concept and can not be mapped to any value other than code of options in openmrs
- FWWOMANYID - 163087AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA is a coded concept and multiselect send value as 1 2 which can not be mapped to this concept
- FWWOMANYID  - 162699AAAAAAAAAAAAAAAAAAAAAAAAAAAAAA concept id missing 3 AAA s
